src/tests/SearchView.test.jsx

```diff
@@ -65,7 +64,7 @@ describe('Search View', () => {
         }),
       ),
     );
-    const user = userEvent.setup();
+    const user = userEvent.setup({ delay: null });
```

Comment:
this was added to prevent timeout when using `async/await` in conjuction with `useFakeTimers`

Reply:
This comment would be better as a real comment in the code (and in all other places where you use this).

Also, have you considered [advanceTimers](https://testing-library.com/docs/user-event/options#advancetimers) as another option? I'm not saying this is better as I don't know much about the library, so I'll leave the choice up to your judgment.

Reply:
I tested this out, and I think it is similar to `runAllTimers` or `runOnlyPendingTimers`. using these (and `advanceTimers`) doesn't prevent the test from timing out (unless maybe I'm using it improperly)

referenced source: https://github.com/testing-library/user-event/issues/565#issuecomment-800694378

Reply:
Thanks for looking at this!

I believe the result is userEvent won't "wait" a small timeout between letters when simulating typing some text in the input. This makes tests a bit more different than what happens in reality, and could hide bugs.
I think it's good to keep it this way for now, there's no need to spend more time on this, but let's just remember that tests are imperfect :-)